dbt/include/trino/macros/materializations/incremental.sql

```diff
@@ -61,43 +130,43 @@
     )
 {%- endmacro %}
 
-{% macro trino__get_merge_sql(target, source, unique_key, dest_columns, predicates=none) -%}
+{% macro trino__get_merge_sql(target, source, unique_key, dest_columns, predicates) -%}
     {%- set predicates = [] if predicates is none else [] + predicates -%}
     {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute="name")) -%}
     {%- set dest_cols_csv_source = dest_cols_csv.split(', ') -%}
-    {%- set update_columns = config.get('merge_update_columns', default = dest_columns | map(attribute="quoted") | list) -%}
+    {%- set merge_update_columns = config.get('merge_update_columns') -%}
+    {%- set merge_exclude_columns = config.get('merge_exclude_columns') -%}
+    {%- set update_columns = get_merge_update_columns(merge_update_columns, merge_exclude_columns, dest_columns) -%}
     {%- set sql_header = config.get('sql_header', none) -%}
 
     {% if unique_key %}
         {% if unique_key is sequence and unique_key is not mapping and unique_key is not string %}
             {% for key in unique_key %}
-                {% set this_key_match -%}
+                {% set this_key_match %}
                     DBT_INTERNAL_SOURCE.{{ key }} = DBT_INTERNAL_DEST.{{ key }}
-                {%- endset %}
+                {% endset %}
                 {% do predicates.append(this_key_match) %}
             {% endfor %}
         {% else %}
-            {% set unique_key_match -%}
+            {% set unique_key_match %}
                 DBT_INTERNAL_SOURCE.{{ unique_key }} = DBT_INTERNAL_DEST.{{ unique_key }}
-            {%- endset %}
+            {% endset %}
             {% do predicates.append(unique_key_match) %}
         {% endif %}
 
         {{ sql_header if sql_header is not none }}
 
-        merge into {{ target }} as DBT_INTERNAL_DEST {{ "\n" }}using {{ source }} as DBT_INTERNAL_SOURCE on
-        {{ predicates | join(' and
-        ') }}
-
-        {%- if unique_key %}
+        merge into {{ target }} as DBT_INTERNAL_DEST
```

Comment:
It may break the SQL formatting, please validate

Reply:
Validated, it is the same as core dbt.